thrust/thrust/system/detail/generic/shuffle.inl

```diff
@@ -69,7 +69,7 @@ class feistel_bijection {
         state[1] = lo & right_side_mask;
     }
     // Combine the left and right sides together to get result
-    return static_cast<std::uint64_t>(state[0] << right_side_bits) | static_cast<std::uint64_t>(state[1]);
+    return (static_cast<std::uint64_t>(state[0]) << right_side_bits) | static_cast<std::uint64_t>(state[1]);
```

Comment:
remark: `right_side_bits` is equal to half of problem size. When problem size exceeds 2^31 elements, we used to shift state out of 32 bits, potentially returning 0 instead of state[0] in high 32 bits of 64 bit value. This change fixes the issue by casting the state to 64-bit value first.